java/dagger/hilt/android/plugin/src/main/kotlin/dagger/hilt/android/plugin/HiltTransformTestClassesTask.kt

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (C) 2020 The Dagger Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dagger.hilt.android.plugin
+
+import com.android.build.gradle.api.ApplicationVariant
+import com.android.build.gradle.api.BaseVariant
+import com.android.build.gradle.api.LibraryVariant
+import com.android.build.gradle.api.UnitTestVariant
+import org.gradle.api.Action
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.Project
+import org.gradle.api.file.FileCollection
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+open class HiltTransformTestClassesTask : DefaultTask() {
+  @InputFiles
+  lateinit var compiledClasses: FileCollection
+
+  @OutputDirectory
+  lateinit var outputDir: File
+
+  @TaskAction
+  @Suppress("UnstableApiUsage")
+  fun transformClasses() {
+    outputDir.deleteRecursively()
+    outputDir.mkdirs()
+
+    val classTransformer = HiltClassTransformer(
+      taskName = name,
+      allInputs = compiledClasses.files.toList()
+    )
+    val convertedDirs = mutableListOf<File>()
+    fun transformClassesInDir(dir: File) {
+      convertedDirs.add(dir)
+      dir.walkTopDown().forEach {
+        if (it.isFile) {
+          val targetOutputDir = outputDir.resolve(
+            it.relativeTo(dir)
+          )
+          classTransformer.transformFile(
+            inputFile = it,
+            outputDir = targetOutputDir.parentFile,
+            outputSourceRootDir = outputDir,
+            skipIfNotHilt = true
+          )
+        }
+      }
+    }
+    // parse the classpath in reverse so that we respect overrides, if it ever happens
+    compiledClasses.files.filter { it.isDirectory }.reversed().forEach {
```

Comment:
the code i'm re-using here is from transform API which does ignore jars.
and yea i just tried w/ 4.1 and it fails, we were using 3.6.3 here. i'll see what we can do there.

Reply:
done. added transform for jars but only here. i'm guessing the regular transform API still receives .class files ?

Reply:
actually seems to be working fine but i'll keep this open in case we should also process contents of jar files for the transform API. FYI we are only interested in files from the compiled module/app.